client/components/banner/test/index.jsx

```diff
@@ -1,165 +1,181 @@
-		const comp = shallow( <Banner { ...props } callToAction={ null } /> );
-		expect( comp.find( '.has-call-to-action' ) ).toHaveLength( 0 );
+		renderWithRedux( <Banner { ...props } dismissPreferenceName="banner-test" /> );
+		expect( screen.getByLabelText( 'Dismiss' ) ).toBeVisible();
 	} );
 
 	test( 'should render a <Button /> when callToAction is specified', () => {
-		const comp = shallow( <Banner { ...props } callToAction={ 'Buy something!' } /> );
-		expect( comp.find( Button ) ).toHaveLength( 1 );
+		render( <Banner { ...props } callToAction="Buy something!" /> );
+		expect( screen.getByRole( 'button' ) ).toHaveTextContent( 'Buy something!' );
 	} );
 
 	test( 'should not render a <Button /> when callToAction is not specified', () => {
-		const comp = shallow( <Banner { ...props } /> );
-		expect( comp.find( Button ) ).toHaveLength( 0 );
+		render( <Banner { ...props } /> );
+		expect( screen.queryByRole( 'button' ) ).not.toBeInTheDocument();
 	} );
 
 	test( 'should have .is-jetpack class and JetpackLogo if jetpack prop is defined', () => {
 		const { plan, ...propsWithoutPlan } = props;
-		const comp = shallow( <Banner { ...propsWithoutPlan } jetpack /> );
-		expect( comp.find( '.is-jetpack' ) ).toHaveLength( 1 );
-		expect( comp.find( 'JetpackLogo' ) ).toHaveLength( 1 );
+		const { container } = render( <Banner { ...propsWithoutPlan } jetpack /> );
+		expect( container.firstChild ).toHaveClass( 'is-jetpack' );
+		expect( container.getElementsByClassName( 'jetpack-logo' ) ).toHaveLength( 1 );
 	} );
 
 	test( 'should render have .is-horizontal class if horizontal prop is defined', () => {
-		const comp = shallow( <Banner { ...props } horizontal /> );
-		expect( comp.find( '.is-horizontal' ) ).toHaveLength( 1 );
+		const { container } = render( <Banner { ...props } horizontal /> );
+		expect( container.firstChild ).toHaveClass( 'is-horizontal' );
 	} );
 
 	test( 'should render a <PlanPrice /> when price is specified', () => {
-		const comp = shallow( <Banner { ...props } price={ 100 } /> );
-		expect( comp.find( PlanPrice ) ).toHaveLength( 1 );
+		render( <Banner { ...props } price={ 100 } /> );
+		expect( screen.queryByTestId( 'banner-price-plan-mock' ) ).toHaveTextContent( '100' );
 	} );
 
 	test( 'should render two <PlanPrice /> components when there are two prices', () => {
-		const comp = shallow( <Banner { ...props } price={ [ 100, 80 ] } /> );
-		expect( comp.find( PlanPrice ) ).toHaveLength( 2 );
+		render( <Banner { ...props } price={ [ 100, 80 ] } /> );
+		expect( screen.queryAllByTestId( 'banner-price-plan-mock' ) ).toHaveLength( 2 );
 	} );
 
 	test( 'should render no <PlanPrice /> components when there are no prices', () => {
-		const comp = shallow( <Banner { ...props } /> );
-		expect( comp.find( PlanPrice ) ).toHaveLength( 0 );
+		render( <Banner { ...props } /> );
+		expect( screen.queryByTestId( 'banner-price-plan-mock' ) ).toBeNull();
```

Comment:
Perhaps we could use `not.toBeInTheDocument()` instead of `toBeNull()`?

Reply:
Just because I can't stop myself from asking questions but I pulled this out of the example you shared earlier: 

https://github.com/Automattic/wp-calypso/blob/9a61a1b494f3af0eaa20d43470a38e2fd8fb1ba9/client/blocks/image-editor/test/image-editor-canvas.jsx#L19

If you went back, would you change that to `not.toBeInTheDocument()` as well (and maybe to `toBeInTheDocument()` or `toBeVisible()`  instead of `toBeDefined()` for the other test)?

Reply:
Yup. I think `.not.toBeInTheDocument()` expresses our intention in a much better way than `toBeNull()` in those scenarios when we want to assert that an element is missing.